lib/cosmos/io/win32_serial_driver.rb

```diff
@@ -112,8 +112,7 @@ def read
 
       loop do
         loop do
-          # Read 1 byte
-          buffer = Win32.read_file(@handle, 1)
+          buffer = Win32.read_file(@handle, @read_max_length - data.length)
```

Comment:
So it reads as much as possible and as the buffer fills it requests less and less until we get all the data?